src/file.c

```diff
@@ -908,6 +908,10 @@ int for_each_file_ex(AL_CONST char *name, int in_attrib, int out_attrib, int (*c
 
    if (!_al_file_isok(name))
       return 0;
+   
+   // Allegro 5 does not return "." or "..", so we must include it ourselves.
+   (*callback)(".", FA_DIREC, param);
+   (*callback)("..", FA_DIREC, param);
 
```

Comment:
https://github.com/liballeg/allegro5/blob/4aa54e6c994af21bc63d8b593673ab3df62390f8/src/fshook_stdio.c#L421

Reply:
I'd have suppressed the ".." if I knew how to check if currently reading the root folder, but I couldn't figure that out

Reply:
I think ".." might be listed even in the root. Just tested on MacOS and it shows ".." existing in the root directory with "ls -a".

Reply:
Also, since `al_find_first()`, etc. are part of the Allegro 4 API, we should probably try to fix this in the driver. This way, users trying to build their Allegro 4 programs get the proper behavior if they are using the `al_find*()` functions.

Reply:
would that look like adding some state to `FF_DATA` to track if `..` and `.` have been "read", and in `al_findnext` filling the `info` param with those values for the first two calls?

Reply:
confirmed that the above approach worked for my use case just as well as the original approach, PR is updated